vl-convert-python/vl_convert.pyi

```diff
@@ -0,0 +1,801 @@
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Any, Literal, TypeAlias
+
+    FormatLocaleName: TypeAlias = Literal[
+        "ar-001",
+        "ar-AE",
+        "ar-BH",
+        "ar-DJ",
+        "ar-DZ",
+        "ar-EG",
+        "ar-EH",
+        "ar-ER",
+        "ar-IL",
+        "ar-IQ",
+        "ar-JO",
+        "ar-KM",
+        "ar-KW",
+        "ar-LB",
+        "ar-LY",
+        "ar-MA",
+        "ar-MR",
+        "ar-OM",
+        "ar-PS",
+        "ar-QA",
+        "ar-SA",
+        "ar-SD",
+        "ar-SO",
+        "ar-SS",
+        "ar-SY",
+        "ar-TD",
+        "ar-TN",
+        "ar-YE",
+        "ca-ES",
+        "cs-CZ",
+        "da-DK",
+        "de-CH",
+        "de-DE",
+        "en-CA",
+        "en-GB",
+        "en-IE",
+        "en-IN",
+        "en-US",
+        "es-BO",
+        "es-ES",
+        "es-MX",
+        "fi-FI",
+        "fr-CA",
+        "fr-FR",
+        "he-IL",
+        "hu-HU",
+        "it-IT",
+        "ja-JP",
+        "ko-KR",
+        "mk-MK",
+        "nl-NL",
+        "pl-PL",
+        "pt-BR",
+        "pt-PT",
+        "ru-RU",
+        "sl-SI",
+        "sv-SE",
+        "uk-UA",
+        "zh-CN",
+    ]
+    TimeFormatLocaleName: TypeAlias = Literal[
+        "ar-EG",
+        "ar-SY",
+        "ca-ES",
+        "cs-CZ",
+        "da-DK",
+        "de-CH",
+        "de-DE",
+        "en-CA",
+        "en-GB",
+        "en-US",
+        "es-ES",
+        "es-MX",
+        "fa-IR",
+        "fi-FI",
+        "fr-CA",
+        "fr-FR",
+        "he-IL",
+        "hr-HR",
+        "hu-HU",
+        "it-IT",
+        "ja-JP",
+        "ko-KR",
+        "mk-MK",
+        "nb-NO",
+        "nl-BE",
+        "nl-NL",
+        "pl-PL",
+        "pt-BR",
+        "ru-RU",
+        "sv-SE",
+        "tr-TR",
+        "uk-UA",
+        "vi-VN",
+        "zh-CN",
+        "zh-TW",
+    ]
+    VegaThemes: TypeAlias = Literal[
+        "carbong10",
+        "carbong100",
+        "carbong90",
+        "carbonwhite",
+        "dark",
+        "excel",
+        "fivethirtyeight",
+        "ggplot2",
+        "googlecharts",
+        "latimes",
+        "powerbi",
+        "quartz",
+        "urbaninstitute",
+        "vox",
+    ]
+    Renderer: TypeAlias = Literal["canvas", "hybrid", "svg"]
+    FormatLocale: TypeAlias = FormatLocaleName | dict[str, Any]
+    TimeFormatLocale: TypeAlias = TimeFormatLocaleName | dict[str, Any]
+    VlSpec: TypeAlias = str | dict[str, Any]

```

Comment:
Though this alias could be a placeholder for if you decided to get more specific in the future.

E.g. if you expect a certain level of nesting, this can be updated and propagate to all relevent defs.

The screenshots in https://github.com/vega/altair/pull/3536 show examples of the utility of this